src/cpp/session/modules/SessionDataImportV2.R

```diff
@@ -13,21 +13,180 @@
 #
 #
 
-.rs.addJsonRpcHandler("preview_data_import", function(path, maxRows, maxCols = 100, maxFactors = 64)
+.rs.addFunction("assemble_data_import_name", function(dataImportOptions)
+{
+  dataName <- dataImportOptions$dataName
+
+  if (is.null(dataName) || identical(dataName, ""))
+    {
+      if (!is.null(dataImportOptions$importLocation))
+      {
+        locationName <- basename(dataImportOptions$importLocation)
+        if (length(locationName) > 0)
+        {
+          fileComponents <- unlist(strsplit(locationName, ".", fixed = TRUE))
+          components <- length(fileComponents)
+          if (components >= 1)
+          {
+            dataName <- paste(fileComponents[1:components-1], collapse = "_")
```

Comment:
In fact:

```R
> n <- 5
> 1:n - 1
[1] 0 1 2 3 4
```

I think you were getting lucky in the previous case as attempting to get the 0th element of an R vector returns an empty vector of the same type.